src/modules/common/helpers/mappings.ts

```diff
@@ -0,0 +1,73 @@
+import { IHomeCarouselData } from "../../home/types/home-types";
+import useMangaStore from "../stores/store";
+import { ITrendingMangaApp } from "../types/app-types";
+
+export const getViewAllCategoryHeading = (category: string) => {
+    switch (category) {
+        case "trending":
+            return "Trending Mangas"
+            break;
+        case "new":
+            return "Latest Mangas"
+            break;
+        default:
+            break;
+    }
+}
+
+export const getCarouselDetails = (category: string): IHomeCarouselData => {
+    switch (category) {
+        case 'new':
+            {
+                return { heading: "Fresh off The Pen", queryKey: "fetchNewMangas" }
+            }
+            break;
+        case 'rank':
+            {
+                return { heading: "Chart-Topping Manga Marvels", queryKey: "fetchTopRankedMangas" }
+            }
+            break;
+        case 'popular':
+            {
+                return { heading: "Tales that Hooked the Hoard", queryKey: "fetchMostFollowedMangas" }
+            }
+            break;
+
+        default:
+            return { heading: "", queryKey: "" }
+            break;
+    }
+}
+
+export const getHomeCarouselState = (category: string) => {
+    switch (category) {
+        case 'new':
+            return useMangaStore.getState().newMangas
+            break;
+        case 'rank':
+            return useMangaStore.getState().topRankedMangas
+            break;
+        case 'popular':
+            return useMangaStore.getState().popularMangas
+            break;
+        default:
+            return []
```

Comment:
I may have missed this but in your code, just make sure to handle these empty return cases gracefully, if you haven't done this already. If you return nothing or break or an empty array, just make sure it has some way of having a default state or feedback to indicate that it is empty/something has gone wrong.

Reply:
The empty array will not harm the flow of the app, instead it won't show the list of mangas on the page. I will include a message indicating that. Thanks a ton for the tip!